AGGREGATE.md

```diff
@@ -1,180 +1,359 @@
+
+However, large domains (like with 2^32 entries) are difficult to process efficiently. This is mitigated by using [incremental distributed point functions](https://github.com/google/distributed_point_functions), which allow us to make “prefix” queries. Rather than returning the full output domain, reporters can ask the processing origins for sums over any bit-prefix of the output domain. This allows us to implement a “hierarchy of histograms”.
+
+For example, at the extreme, if you have only 4 aggregation keys you want to measure, you could use the lowest order 2 bits in your 32 bit aggregation key space, and query just on that.
+
+This scheme could also support dynamic re-aggregation where the same data is evaluated multiple times over multiple levels of a hierarchy to learn a sparse domain. This is aided by the ability to evaluate only a portion of the domain which helps for large domain sizes. This could even be done by the processing origins themselves, which can jointly work to evaluate a hierarchy, and return the hierarchy back to the reporter.  It is also possible to enforce restrictions on the portion of the domain queried, which could improve privacy in some ways (e.g. so you couldn’t have a bucket-per-user).
+
+**Note:** as new and improved protocols are invented, we can consider adding them to the API in some form, this includes other aggregation functions like using larger domains (> 2^32) with some thresholding mechanism.
+
+**Note:** As mentioned in the [SERVICE.md](https://github.com/WICG/conversion-measurement-api/blob/main/SERVICE.md) doc, we are planning on adding differential privacy noise to the output of the MPC computation. For hierarchical computation, this will require adding noise to each level.
+
+### Example query model
+
+The following is a high level query model that could be achieved with the incremental distributed point function (DPF) primitive.
+
+Imagine the following prefix-tree of counts, where each key is 3 bits long in total (last level of the tree)
 ```
-conversion-signing-url=<URL>
+1-bit prefix counts:       102                   402
+2-bit prefix counts:    99       3           400       2
+3-bit prefix counts: 99   0    1   2      200   200   1  1
 ```
 
-Once an impression / conversion is registered, the browser should interact with the signing endpoint, which should provide authentication tokens that will attach to conversion reports and be verified by the helper servers. Helper servers can (optionally) discard reports that are not signed. This is explored in more detail [here](SERVICE.md#authenticating-inputs).
 
-### Many-per-click / Once-per-click
+We could have a query configuration asking for counts at every prefix level, with some strategy for iterating down and pruning the tree.
+
+
+```json
```

Comment:
ditto to below on highlighting

Could also consider dropping json for this one

Reply:
done